### ISSUE TITLE
Design New Modules for Refactoring Legacy Code

### ISSUE THREAD

**Issue by tatmaca** [uchicago-cs/chiventure#1225]
Team Members: Togay & Ramarko

Description: During the last custom-scripts team meeting, we decided that we should build on top of the previously established legacy code from 2 years ago. Before this, however, we decided that it would be best to refactor the code such that it would be easier for both us and other development teams to understand how the implementation of our custom scripts will work.

This task is complete when we have fully refactored the legacy code into an appropriate amount of modules such that all the operations remain the same and the code is easier to read.

**Comment by tatmaca** [comment]
The custom-scripts team has refactored the existing legacy code into four new modules. After reviewing the code, we decided that we could refactor the source code into four modules: the arguments (arg) module, the object_type (obj) module, the lua module, and the implement into game (get) module. 

We decided that the "arg" and "obj" modules would be necessary as it stored the creation and initialization of the two main structs used in the custom-scripts source code. We decided that the "lua" module was necessary as it included all of the unique functions that allowed our code to interact with Lua. We decided that the "get" module was necessary as it stores all the code that members of other developments teams would be interested in looking at in order to get their object_types to interact with their unique custom scripts.

As such, this meets the requirements for the completion of this Issue.
